scb-engine/src/main/java/io/securecodebox/engine/auth/CamundaAuthenticationProvider.java

```diff
@@ -25,6 +30,17 @@ public Authentication authenticate(Authentication auth) throws AuthenticationExc
         boolean authenticated = engine.getIdentityService().checkPassword(username,password);
 
         if (authenticated) {
+            // Set current camunda authentication
+            User user = engine.getIdentityService().createUserQuery().userId(username).singleResult();
+            List<String> groupIds = engine.getIdentityService()
+                    .createGroupQuery()
+                    .groupMember(username)
+                    .list()
+                    .stream()
+                    .map(Group::getId)
+                    .collect(Collectors.toList());
+            engine.getIdentityService().setAuthentication(user.getId(), groupIds);
```

Comment:
Sets authentication for the current thread.
Depending on spring implementation will likely need to call `clearAuthentication()` after the request is done, to ensure that we cannot poison the next request with the previous credentials.

Reply:
Alternatively we could leave the camunda authentication unset and access the current user via the spring authentication. This would probably be simpler than this.